pkg/agent/agent.go

```diff
@@ -484,3 +499,87 @@ func sleepOrDone(d time.Duration, done <-chan struct{}) {
 		return
 	}
 }
+
+// splitNewlineEnv splits newline-delimited KEY=VAL pairs and update map.
+func splitNewlineEnv(m map[string]string, envs string) {
+	sc := bufio.NewScanner(strings.NewReader(envs))
+	for sc.Scan() {
+		spl := strings.SplitN(sc.Text(), "=", 2)
+
+		// Just skip empty lines or lines without a value.
+		if len(spl) == 1 {
+			continue
+		}
+
+		m[spl[0]] = spl[1]
+	}
+}
+
+// versionInfo contains Flatcar version and update information.
+type versionInfo struct {
+	id      string
+	group   string
+	version string
+}
+
+func getUpdateMap() (map[string]string, error) {
+	infomap := map[string]string{}
+
+	// This file should always be present on Flatcar.
+	b, err := ioutil.ReadFile(updateConfPath)
```

Comment:
instead of `ReadFile` we can open the file:
```suggestion
	b, err := os.Open(updateConfPath)
```
the file handler implements `Read` method, so we should be able to pass it directly to `bufio.NewScanner(b)` without intermediary string conversion

Reply:
Interesting, thanks! I wonder how this could work with the overrides. I guess we could aggregate `io.Reader` somehow and just make `splitNewlineEnv` to accept an `io.Reader` and return a `map[string]string`, instead of modifying given one.

However, we can do this also when we have some tests for agent code :sweat_smile: